### PR TITLE
Checkdoublevoteentry

### PR DESCRIPTION
**Pull request by xt-t**
Option to select Restaurants from Favourites implemented

### CODE REVIEW

File: backend/src/main/java/de/neuefische/backend/controller/TastyDateController.java [hunk 1]
```diff
@@ -15,12 +16,25 @@
...
-    private final DateSettingsService dateInfoService;
+    private final TastyDateSettingsService dateInfoService;
 
-    public TastyDateController(DateSettingsService dateInfoService) {
+    public TastyDateController(TastyDateSettingsService dateInfoService) {
         this.dateInfoService = dateInfoService;
     }
 
+
+    @GetMapping("/getAllTastyDateItems")
```
Comment (Flooooooooooorian) [inline backend/src/main/java/de/neuefische/backend/controller/TastyDateController.java:26]: Try to stick why the naming conventions 👍 https://restfulapi.net/resource-naming/

```suggestion
    @GetMapping
```

File: backend/src/main/java/de/neuefische/backend/controller/TastyDateController.java [hunk 2]
```diff
@@ -15,12 +16,25 @@
...
+        return ok(allTastyDateItems);
+    }
+
+    @GetMapping(value="/{tastyDateId}")
+    public ResponseEntity<TastyDateItem> getTastyDateById  (@PathVariable String tastyDateId) {
+        Optional<TastyDateItem> opt = dateInfoService.findTastyDateById(tastyDateId);
+        return ResponseEntity.of(opt);
+    }
+
     @PostMapping("/completesettings")
```
Comment (Flooooooooooorian) [inline backend/src/main/java/de/neuefische/backend/controller/TastyDateController.java:38]:
```suggestion
    @PostMapping
```

File: backend/src/main/java/de/neuefische/backend/model/settingsSubModel/appointmentsettings/RestaurantItem.java [hunk 3]
```diff
@@ -11,6 +11,7 @@
     public class RestaurantItem {
 
         private int id;
+        private String creatorName;
```
Comment (Flooooooooooorian) [inline backend/src/main/java/de/neuefische/backend/model/settingsSubModel/appointmentsettings/RestaurantItem.java:14]:
```suggestion
        private String creatorUsername;
```

File: backend/src/main/java/de/neuefische/backend/service/TastyDateSettingsService.java [hunk 4]
```diff
@@ -27,28 +26,28 @@ public TastyDateItem addDateSettings(TastyDateItem settingsItem) {
...
+        return dateSettingsRepo.findAll();
+    }
 
-        //Optional - existiert das SettingsItem? Fehlermeldung
+    public Optional<TastyDateItem> findTastyDateById(String id) {return dateSettingsRepo.findById(id);}
 
+    public TastyDateItem addVoteTimeItemToTastyDate(UserTimeVote timeVote, String tastyDateId) {
         //checking and updating the List of the UserVotes for the DateTimes
         Optional<TastyDateItem> optionalTastyDateItem = dateSettingsRepo.findById(tastyDateId);
         TastyDateItem test = optionalTastyDateItem.orElseThrow(() -> new NoSuchElementException("TastyDateItem with id: " + tastyDateId + " does not exists!"));
```
Comment (Flooooooooooorian) [inline backend/src/main/java/de/neuefische/backend/service/TastyDateSettingsService.java:38]: "test" better find another variable name

File: backend/src/main/java/de/neuefische/backend/service/TastyDateSettingsService.java [hunk 5]
```diff
@@ -75,12 +74,8 @@ public TastyDateItem addVoteRestaurantItemToTastyDate(UserRestaurantVote restaur
...
             boolean[] checksItem = item.getVotesPerRestaurantFromOneUser();
             for (int i=0; i < checksVoteItem.length; i++) {
-                if (checksItem[i]) {
-                    positiveAmount[i]++;
-                }
-                else {
-                    negativeAmount[i]++;
-                }
+                if (checksItem[i]) positiveAmount[i]++;
+                else negativeAmount[i]++;
```
Comment (Flooooooooooorian) [inline backend/src/main/java/de/neuefische/backend/service/TastyDateSettingsService.java:87]: In java you typically use the "classic"  code stxyle for if statements
```
if (condition) {
    ...
}
```

File: frontend/src/components/appointmentsettings/AppointThree/AppointThreeAddFavouriteRestaurants.tsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+interface AppointThreeAddFavouriteRestaurantsProps {
+    appointThree: AppointThreeType
+    resetDataInput: Function
+}
+
+export default function AppointThreeAddFavouriteRestaurants({
+                                                              appointThree,
+                                                              resetDataInput
+                                                          }: AppointThreeAddFavouriteRestaurantsProps) {
+    const [open, setOpen] = useState(false);
```
Comment (Flooooooooooorian) [inline frontend/src/components/appointmentsettings/AppointThree/AppointThreeAddFavouriteRestaurants.tsx:26]: I would put all the state to the top, underneath useeffects and all other functions and then the return statement

File: frontend/src/components/appointmentsettings/AppointTwo/TimeSetter.tsx [hunk 7]
```diff
@@ -5,51 +5,26 @@ import LocalizationProvider from '@mui/lab/LocalizationProvider';
 import TimePicker from '@mui/lab/TimePicker';
 
 interface TimeSetterProps {
-    startTime: Date | null
-    setStartTime: Function
-    endTime: Date | null
-    setEndTime: Function
-    range: string
+    timeInput: Date | null
```
Comment (Flooooooooooorian) [inline frontend/src/components/appointmentsettings/AppointTwo/TimeSetter.tsx:8]: I would only allow Date as a type or is there a case where a TastyDate has no Date? 🤔

File: frontend/src/components/appointmentsettings/AppointTwo/AppointTwoCardAddDate.tsx [hunk 8]
```diff
@@ -27,19 +39,15 @@ export default function AppointTwoCardAddDate(
...
 
                     <TimeSetter
-                        range="start Time"
-                        startTime={startTime}
-                        setStartTime={setStartTime}
-                        endTime={endTime}
-                        setEndTime={setEndTime}
+                        timeInput={startTime}
+                        setTimeInput={setStartTime}
+                        testIfTimeRestrictionAvailable={noTimeRestriction}
```
Comment (Flooooooooooorian) [inline frontend/src/components/appointmentsettings/AppointTwo/AppointTwoCardAddDate.tsx:44]: Why do you pass the function here and not the result directly? And i would make the prop "testIfTimeRestrictionAvailable"/"minTime" optional so you can leave it out when you dont need it

File: frontend/src/pages/result/OverviewTastyDateItems.tsx [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    const [selectedTastyDateId, setSelectedTastyDateId]=useState<string>("");
+    const [tastyDateItems, setTastyDateItems]=useState<TastyDateItem[]>([]);
+
+    const navigate = useNavigate();
+
+    const switchPageToCertainVote = () => {
+        navigate(`/overview/${selectedTastyDateId}`);
+    }
+
+    const getEveryTastyDateId = () => {
```
Comment (Flooooooooooorian) [inline frontend/src/pages/result/OverviewTastyDateItems.tsx:30]:
```suggestion
    const getEveryTastyDate = () => {
```

File: frontend/src/pages/result/VoteResult.tsx [hunk 10]
```diff
@@ -1,51 +1,87 @@
...
     const [userName, setUserName] = useState<string>("");
-    const [tempName, setTempName] = useState<string>("");
+    const [checkIfNameConfirmed, setCheckIfNameConfirmed] = useState<boolean>(false);
 
     const handleChange = (event: React.SyntheticEvent, newValue: number) => {
-        setValue(newValue);
+        setVoteType(newValue);
     };
 
+    if (typeof tastyDateItemForVote === "object") {
```
Comment (Flooooooooooorian) [inline frontend/src/pages/result/VoteResult.tsx:46]: Why do you check for type? Is there a possiblity that you recive something else than a object? I would check for undefined

**Review by Flooooooooooorian — APPROVED** [review]

File: backend/src/main/java/de/neuefische/backend/controller/TastyDateController.java (hunk 1 above)
Comment (xt-t) [reply backend/src/main/java/de/neuefische/backend/controller/TastyDateController.java:26]: I checked the hyperlink and altered the names of my REST API

File: backend/src/main/java/de/neuefische/backend/controller/TastyDateController.java (hunk 2 above)
Comment (xt-t) [reply backend/src/main/java/de/neuefische/backend/controller/TastyDateController.java:38]: See my last comment

File: backend/src/main/java/de/neuefische/backend/service/TastyDateSettingsService.java (hunk 4 above)
Comment (xt-t) [reply backend/src/main/java/de/neuefische/backend/service/TastyDateSettingsService.java:38]: Found one and changed the name!

File: backend/src/main/java/de/neuefische/backend/service/TastyDateSettingsService.java (hunk 5 above)
Comment (xt-t) [reply backend/src/main/java/de/neuefische/backend/service/TastyDateSettingsService.java:87]: Alright, I just assumed that it is a cleaner coding style.

File: frontend/src/components/appointmentsettings/AppointThree/AppointThreeAddFavouriteRestaurants.tsx (hunk 6 above)
Comment (xt-t) [reply frontend/src/components/appointmentsettings/AppointThree/AppointThreeAddFavouriteRestaurants.tsx:26]: OK! I try to pay attention to it in the future. Of course I also changed it.

File: frontend/src/pages/result/OverviewTastyDateItems.tsx (hunk 9 above)
Comment (xt-t) [reply frontend/src/pages/result/OverviewTastyDateItems.tsx:30]: OK. The naming was obviously confusing. I changed it.

File: backend/src/main/java/de/neuefische/backend/model/settingsSubModel/appointmentsettings/RestaurantItem.java (hunk 3 above)
Comment (xt-t) [reply backend/src/main/java/de/neuefische/backend/model/settingsSubModel/appointmentsettings/RestaurantItem.java:14]: File has been deleted

File: frontend/src/pages/result/VoteResult.tsx (hunk 10 above)
Comment (xt-t) [reply frontend/src/pages/result/VoteResult.tsx:46]: Implemented!

File: frontend/src/components/appointmentsettings/AppointTwo/AppointTwoCardAddDate.tsx (hunk 8 above)
Comment (xt-t) [reply frontend/src/components/appointmentsettings/AppointTwo/AppointTwoCardAddDate.tsx:44]: Done!

File: frontend/src/components/appointmentsettings/AppointTwo/TimeSetter.tsx (hunk 7 above)
Comment (xt-t) [reply frontend/src/components/appointmentsettings/AppointTwo/TimeSetter.tsx:8]: After a review session with you, we successfully changed it!

**Pull request merged**